internal/services/containerapps/container_app_resource_test.go

```diff
@@ -1666,6 +1702,80 @@ resource "azurerm_container_app" "test" {
 `, r.template(data), data.RandomInteger)
 }
 
+func (r ContainerAppResource) ingressSecurityRestriction(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_container_app" "test" {
+  name                         = "acctest-capp-%[2]d"
+  resource_group_name          = azurerm_resource_group.test.name
+  container_app_environment_id = azurerm_container_app_environment.test.id
+  revision_mode                = "Single"
+
+  template {
+    container {
+      name   = "acctest-cont-%[2]d"
+      image  = "jackofallops/azure-containerapps-python-acctest:v0.0.1"
+      cpu    = 0.25
+      memory = "0.5Gi"
+    }
+  }
+
+  ingress {
+    target_port = 5000
+    ip_security_restriction = [
+      {
+        name             = "test"
+        description      = "test"
+        action           = "Allow"
+        ip_address_range = "0.0.0.0/0"
+      }
+    ]
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r ContainerAppResource) ingressSecurityRestrictionUpdate(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_container_app" "test" {
+  name                         = "acctest-capp-%[2]d"
+  resource_group_name          = azurerm_resource_group.test.name
+  container_app_environment_id = azurerm_container_app_environment.test.id
+  revision_mode                = "Single"
+
+  template {
+    container {
+      name   = "acctest-cont-%[2]d"
+      image  = "jackofallops/azure-containerapps-python-acctest:v0.0.1"
+      cpu    = 0.25
+      memory = "0.5Gi"
+    }
+  }
+
+  ingress {
+    target_port = 5000
+    ip_security_restriction = [
+      {
+        name             = "test"
+        description      = "test"
+        action           = "Allow"
+        ip_address_range = "0.0.0.0/0"
+      },
+      {
+        name             = "test2"
+        description      = "test2"
+        action           = "Deny"
+        ip_address_range = "10.0.0.0/8"
+      }
+    ]
```

Comment:
as above:
```suggestion
    ip_security_restriction {
      name             = "test"
      description      = "test"
      action           = "Allow"
      ip_address_range = "10.1.0.0/16"
    }

    ip_security_restriction {
      name             = "test2"
      description      = "test2"
      action           = "Allow"
      ip_address_range = "10.2.0.0/16"
    }

    traffic_weight {
      latest_revision = true
      percentage      = 100
    }
```
Also note, the rules must either all be `Allow` or all be `Deny`, the API will reject mixed rules.